packages/lib/src/getClosestFocusableElement/getClosestFocusableElement.tsx

```diff
@@ -0,0 +1,32 @@
+const focusableSelectors = [
+  'a',
+  'button',
+  'frame',
+  'iframe',
+  'input:not([type=hidden])',
+  'select',
+  'textarea',
+  '*[tabindex]',
+] as const;
+
+/**
+ * Crawls up the DOM tree to find the closest focusable element.
+ * Returns the element itself if it is focusable.
+ * If no focusable element is found, it returns the body element.
+ *
+ * This is useful for ensuring that focus is returned to a valid element
+ * after an event, such as a click or key press.
+ */
+export const getClosestFocusableElement = (el: HTMLElement): HTMLElement => {
+  // check if the element is focusable
+  if (el.matches(focusableSelectors.join(','))) {
```

Comment:
Would [closest](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest) do the same thing? It would return the nearest ancestor, including itself? And then you can prob avoid recursion.

Reply:
Oh, yeah. Last time I looked at that it wasn't well supported